### DISCUSSION TITLE
TextChannel.send with custom username & avatar?

### DISCUSSION THREAD

**Discussion by GamePlayer-8** [Rapptz/discord.py#9202] [Q&A] [answered]
Heyo,

I can't find in the [API Reference](https://discordpy.readthedocs.io/en/stable/api.html?highlight=textchannel%20send#discord.TextChannel.send) information about setting custom username & avatar for a message. Is there any way to do it?

**Comment by Rapptz** [comment]
That feature is only available for Webhooks so it'd be for [`Webhook.send`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Webhook.send).

**Comment by GamePlayer-8** [comment, marked as answer]
So using `discord.py` + `requests` libraries & a script:
```python
import aiohttp
import discord
from discord import Webhook

intents = discord.Intents.all()
intents.members = True
bot = commands.Bot(command_prefix=PREFIX, intents=intents)

@bot.event
async def on_message(message):
    if message.author == bot.user:
        return
    async with aiohttp.ClientSession() as session:
        webhook = Webhook.from_url('webhook_url', session=session)
        await webhook.send(content=message.content, username=message.author.name, avatar_url=message.author.avatar.url)
```

is the solution.

**Reply by Rapptz** [reply]
Please use the asynchronous version of the webhook with `aiohttp`. Blocking code is an issue as detailed in the FAQ.

**Reply by GamePlayer-8** [reply]
There's an issue with `AsyncWebhookAdapter` which I can't import for the typical webhook.
Did you mean, adding line `async with aiohttp.ClientSession() as session:`?

**Reply by Rapptz** [reply]
https://discordpy.readthedocs.io/en/stable/migrating.html#major-webhook-changes

**Reply by GamePlayer-8** [reply]
Alr, fixed. 👍🏻
